lib/ex_unit/lib/ex_unit/case.ex

```diff
@@ -532,15 +550,109 @@ defmodule ExUnit.Case do
+
+        describe "using custom attribute" do
+          @describe_fixtures :user
+          @describe_fixtures {:post, insert: false}
+
+          test "has attribute", context do
+            assert context.registered.describe_fixtures == [{:post, insert: false}, :user]
+          end
+        end
+
+        describe "custom attributes are cleared per describe" do
+          test "doesn't have attributes", context do
+            assert context.registered.describe_fixtures == []
+          end
+        end
+      end
+
+  """
+  @spec register_describe_attribute(env, name :: atom, opts :: keyword) :: :ok
+  def register_describe_attribute(env, name, opts \\ []) do
+    do_register_attribute(:ex_unit_registered_describe_attributes, env, name, opts)
+  end
+
+  @doc """
+  Registers a new module attribute to be used during `ExUnit.Case` tests.
+
+  The attribute values will be available through `context.registered`.
+
+  `Module.register_attribute/3` is used to register the attribute,
+  this function takes the same options.
+
+  ## Examples
+
+      defmodule MyTest do
+        use ExUnit.Case
+
+        ExUnit.Case.register_module_attribute(__MODULE__, :module_fixtures, accumulate: true)
+
+        @module_fixtures :user
+        @module_fixtures {:post, insert: false}
+
+        test "using custom attribute", context do
+          assert context.registered.fixtures == [{:post, insert: false}, :user]
+        end
+
+        test "still using custom attribute", context do
+          assert context.registered.fixtures == [{:post, insert: false}, :user]
+        end
+      end
+
+  """
+  @spec register_module_attribute(env, name :: atom, opts :: keyword) :: :ok
+  def register_module_attribute(env, name, opts \\ []) do
+    do_register_attribute(:ex_unit_registered_module_attributes, env, name, opts)
+  end
 
-  def register_attribute(%{module: module}, name, opts) do
-    register_attribute(module, name, opts)
+  defp do_register_attribute(type, %{module: module}, name, opts) do
+    do_register_attribute(type, module, name, opts)
   end
 
-  def register_attribute(mod, name, opts) when is_atom(mod) and is_atom(name) and is_list(opts) do
+  defp do_register_attribute(type, mod, name, opts)
+       when is_atom(type) and is_atom(mod) and is_atom(name) and is_list(opts) do
+    validate_registered_attribute!(type, mod, name)
     Module.register_attribute(mod, name, opts)
-    Module.put_attribute(mod, :ex_unit_registered, name)
+    Module.put_attribute(mod, type, name)
+  end
+
+  defp validate_registered_attribute!(type, mod, name) do
+    registered_attribute_keys = [
+      :ex_unit_registered_module_attributes,
+      :ex_unit_registered_describe_attributes,
+      :ex_unit_registered_test_attributes
+    ]
+
+    for key <- registered_attribute_keys,
+        type != key && name in Module.get_attribute(mod, key) do
+      raise "cannot register attribute #{inspect(name)} multiple times"
```

Comment:
Personally, I think this is a problem with the argument you gave to this function, so I'd raise an `ArgumentError` here.

```suggestion
      raise ArgumentError, "cannot register attribute #{inspect(name)} multiple times"
```